src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/RequestExecutionQueue.cs

```diff
@@ -196,13 +243,36 @@ private async Task ProcessQueueAsync()
                         // will be sent back to the client but they can also be captured via HandleNonMutatingRequestError,
                         // though these errors don't put us into a bad state as far as the rest of the queue goes.
                         // Furthermore we use Task.Run here to protect ourselves against synchronous execution of work
-                        // blocking the request queue for longer periods of time (it enforces parallelizabilty).
-                        _ = WrapStartRequestTaskAsync(Task.Run(() => work.StartRequestAsync(context, cancellationToken), cancellationToken), rethrowExceptions: false);
+                        // blocking the request queue for longer periods of time (it enforces parallelizability).
+                        var currentWorkTask = WrapStartRequestTaskAsync(Task.Run(() => work.StartRequestAsync(context, cancellationToken), cancellationToken), rethrowExceptions: false);
+
+                        if (CancelInProgressWorkUponMutatingRequest)
+                        {
+                            if (currentWorkCts is null)
+                            {
+                                throw new InvalidOperationException($"unexpected null value for {nameof(currentWorkCts)}");
+                            }
```

Comment:
```suggestion
                            Contract.ThrowIfNull(currentWorkCts);
```

Reply:
IIRC Contract comes out of `Roslyn.Utilities`? If so we can't do this without adding a dependency on Roslyn dll's, which we want to avoid.

Reply:
We can just link in files we want.  ideally, these would all be in a source-package anyways, so we could jsut link that source-package.  But i'm not sure if we're there yet.

We should come up with *Some* solution here, otherwise this library is in roslyn, but just doesn't benefit from teh wealth of helpers/libs we have :)

Reply:
The Contract class also utilizes nullability attributes, so should I enable that in this project too?

Reply:
100% yes.